File: 1624/solution.go
package main

func maxLengthBetweenEqualCharacters(s string) int {
	res := -1
	locs := make(map[byte]int)
	for i := 0; i < len(s); i++ {
		char := s[i]
		loc, ok := locs[char]
		if ok {
			res = max(res, i-loc-1)
		} else {
			locs[char] = i
		}
	}
	return res
}

func max(a int, b int) int {
	if a >= b {
		return a
	}
	return b
}

File: 1624/solution.py
class Solution:
    def maxLengthBetweenEqualCharacters(self, s: str) -> int:
        locs = {}
        for i, v in enumerate(s):
            if v in locs:
                locs[v][1] = i
            else:
                locs[v] = [i, i]
        # res = -1
        # for start, end in locs.values():
        #     res = max(res, end - start - 1)
        # return res

        return max([end - start - 1 for start, end in locs.values()])
        # time O(n)
        # space O(1)
